scripts/bdc/run_dbgap_xml_gen_fallback.py

```diff
@@ -0,0 +1,471 @@
+            summary_content.append(
+                f"  {i}. {study['study_id']} ({study['accession_id']}) - Reason: {study['details']}"
+            )
+    
+    if not failed.empty:
+        summary_content.append("\nFailed studies:")
+        for i, (_, study) in enumerate(failed.iterrows(), 1):
+            summary_content.append(
+                f"  {i}. {study['study_id']} ({study['accession_id']}) - Reason: {study['details']}"
+            )
+    
+    # Write to a separate summary file
+    summary_txt_path = os.path.join(timestamp_dir, 'processing_summary.txt')
+    with open(summary_txt_path, 'w') as f:
+        for line in summary_content:
+            f.write(line + "\n")
+    for line in summary_content:
+        logging.info(line)
+        
+    return summary_txt_path
+
+def get_program_dir(gen3_metadata):
+    # Extract program from metadata
+    if isinstance(gen3_metadata, pd.Series):
+        program = gen3_metadata.get('Program', '')
+    else:
+        program = gen3_metadata.get('Program', '')
+    
+    # Clean program name for directory use
+    if not program or pd.isna(program):
+        program = 'unknown_program'
+    else:
+        # Handle multi-program case (pipe separated in Gen3)
+        if '|' in program:
+            program = program.split('|')[0]  # Use first program
+        
+        # Clean program name for directory use
+        program = program.strip().replace(' ', '_').replace('/', '_').lower()
+    
+    return program
+
+def cleanup_empty_directory(directory):
+    try:
+        if os.path.exists(directory):
+            has_content = False
+            for root, dirs, files in os.walk(directory):
+                if files:
+                    has_content = True
+                    break
+            
+            if not has_content:
+                shutil.rmtree(directory)
+                logging.info(f"Removed empty directory: {directory}")
+    except Exception as e:
+        logging.warning(f"Failed to clean up directory {directory}: {str(e)}")
+
+def process_study(study_id, dbgap_id, study_name, output_dir, config, summary_df):
+    """Process a single study with download attempt and fallback"""
+    process_msg = f"\nProcessing study: {study_id} (dbGaP: {dbgap_id})"
+    logging.info(process_msg)
+    
+    # Get study row from Gen3 metadata to extract program
+    try:
+        gen3_df = pd.read_csv(config["GEN3_CSV"])
+        gen3_df['study_id'] = gen3_df['Accession'].str.split('.').str[0]
+        study_row = gen3_df[gen3_df['study_id'] == study_id].iloc[0]
+        program_name = get_program_dir(study_row)
+        
+        # Create program directory
+        program_dir = os.path.join(output_dir, program_name)
+        os.makedirs(program_dir, exist_ok=True)
+        
+        # Update study directory to be within program directory
+        study_dir = os.path.join(program_dir, dbgap_id)
+    except (IndexError, KeyError, Exception) as e:
```

Comment:
```suggestion
    except Exception as e:
```

I think IndexError and KeyError are both subclasses of Exception, so you don't need to include those unless it's specifically to be clearer about these classes.

Reply:
only benefit of explicitly listing IndexError and KeyError is clarity when reading the code - it makes it more obvious what specific errors are expected to occur and will remove Exception